### PR TITLE
[WPE][cross-toolchain-helper] Add chromium to webkitdevci images, port to scarthgap yocto release

### PR DESCRIPTION
**Pull request by nikolaszimmermann**
#### 9c523c665516fbdce6dc4cfe2a28f8451fe7d6da
<pre>
[WPE][cross-toolchain-helper] Add chromium to webkitdevci images, port to scarthgap yocto release
<a href="https://bugs.webkit.org/show_bug.cgi?id=275846">https://bugs.webkit.org/show_bug.cgi?id=275846</a>

Reviewed by Carlos Alberto Lopez Perez.

- Port rpi/riscv targets to latest yocto release (scarthgap)
- Include Chromium in the webkitdevci image for the rpi targets, for performance comparisions.
  Therefore we need a recent clang version - the mickledore yocto release is not sufficient.
  Furthermore we want to build chromium in the most &quot;fair&quot; way, not using some GCC flavour,
  but strictly following upstream recommendations, to end up with an unbiased comparision.

Covered by CI test infrastructure.

* Tools/yocto/meta-openembedded_and_meta-webkit.patch: Added.
* Tools/yocto/meta-openembedded_backport-libbacktrace-and-newer-netdata-to-mickledore-branch.patch: Removed.
* Tools/yocto/riscv/manifest.xml:
* Tools/yocto/rpi/bblayers.conf:
* Tools/yocto/rpi/local-rpi3-32bits-mesa.conf:
* Tools/yocto/rpi/local-rpi3-32bits-userland.conf:
* Tools/yocto/rpi/local-rpi3-64bits-mesa.conf:
* Tools/yocto/rpi/local-rpi4-32bits-mesa.conf:
* Tools/yocto/rpi/local-rpi4-64bits-mesa.conf:
* Tools/yocto/rpi/manifest.xml:
* Tools/yocto/targets.conf:

Canonical link: <a href="https://commits.webkit.org/280438@main">https://commits.webkit.org/280438@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/9ff9ac00ebe33abd75f446948433a28edc817677

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56623 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35950 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9096 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60237 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7064 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58750 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43573 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7255 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45871 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4945 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58653 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33798 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48875 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26731 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30578 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6206 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6067 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52530 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6477 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61920 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/535 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6574 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/53127 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/537 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48933 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53137 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12531 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/463 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31778 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32864 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33948 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32610 "Built successfully") | | |

### CODE REVIEW

**Comment by nikolaszimmermann** [comment]
Still a draft, need to test this.

**Comment by nikolaszimmermann** [comment]
NOTE: The netdata apps_groups.conf has been migrated to the repo which hosts the buildbot/ews scripts for the WebKit Perf bots, doesn't need to be part of the yocto definitions.

File: Tools/yocto/targets.conf [hunk 1]
```diff
@@ -86,5 +86,4 @@ conf_bblayers_path = riscv/bblayers.conf
 conf_local_path = riscv/local-qemu-riscv64.conf
 image_basename = webkit-dev-ci-tools
 image_types = tar.xz wic.xz wic.bmap
-patch_file_path = meta-openembedded_backport-libbacktrace-and-newer-netdata-to-mickledore-branch.patch
```
Comment (clopez) [inline Tools/yocto/targets.conf:89]: doesn't this target also need the patch?

Comment (nikolaszimmermann) [reply Tools/yocto/targets.conf:89]: You're right, will adapt!

Comment (nikolaszimmermann) [reply Tools/yocto/targets.conf:89]: I didn't port to riscv target to the new yocto release at all. Do you want to me do that?

Comment (nikolaszimmermann) [reply Tools/yocto/targets.conf:89]: Ahh, I forgot to remove the riscv/manifests.xml from the patch, but you're right, makes sense to migrate both to the new yocto release, will fix that.

**Comment by nikolaszimmermann** [comment]
I didn't include chromium for the riscv target, not needed there, but properly ported all targets to scarthgap.

**Comment by clopez** [comment]
> I didn't include chromium for the riscv target, not needed there, but properly ported all targets to scarthgap.

yes, that make sense.
Also is likely it will give build issues on that target.

File: Tools/yocto/rpi/bblayers.conf [hunk 2]
```diff
@@ -5,6 +5,9 @@ POKY_BBLAYERS_CONF_VERSION = "2"
 BBPATH = "${TOPDIR}"
 BBFILES ?= ""
 
+# Needed for systemd-analyze
+DISTRO_FEATURES:append = " usrmerge"
+
```
Comment (clopez) [inline Tools/yocto/rpi/bblayers.conf]: This is not the best place for it.
I suggest to put this instead in meta-webkit/conf/distro/webkitdevci.conf (via a patch) and also send a PR to meta-webkit

File: Tools/yocto/meta-openembedded_and_meta-webkit.patch [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+index e5223cc..1454b62 100644
+--- a/sources/meta-openembedded/meta-python/recipes-devtools/python/python3-twisted_24.3.0.bb
++++ b/sources/meta-openembedded/meta-python/recipes-devtools/python/python3-twisted_24.3.0.bb
+@@ -15,7 +15,7 @@ do_install:append() {
+     find ${D} \( -name "*.bat" -o -name "*.c" -o -name "*.h" \) -exec rm -f {} \;
+ }
+ 
+-PACKAGES += "\
++PACKAGES =+ "\
+     ${PN}-zsh \
```
Comment (clopez) [inline Tools/yocto/meta-openembedded_and_meta-webkit.patch:11]: Is this change needed?
The string appended already starts by an space (several actually)

Comment (nikolaszimmermann) [reply Tools/yocto/meta-openembedded_and_meta-webkit.patch:11]: Yeah, I cherry-picked that patch from 'main' branch (not in scarthgap branch at present). Otherwise python3-twisted fails..

File: Tools/yocto/rpi/bblayers.conf (hunk 2 above)
Comment (nikolaszimmermann) [reply Tools/yocto/rpi/bblayers.conf]: Done (except for upstreaming). Will postpone upstreaming the meta-oe and meta-webkit patches, until I learned how to contribute their :-) Noted on my TODO list.

Comment (nikolaszimmermann) [reply Tools/yocto/rpi/bblayers.conf]: Ok, that was easy, just made it:
https://github.com/Igalia/meta-webkit/pull/502
https://github.com/openembedded/meta-openembedded/pull/838

**Comment by nikolaszimmermann** [comment]
Worked: `NOTE: Tasks Summary: Attempted 12163 tasks of which 12074 didn't need to be rerun and all succeeded.`

**Comment by clopez** [comment]
Thanks! 
Let's try this

**Review by clopez — APPROVED** [review]
